### PR TITLE
Bringup BLOOM-176B model

### PR DESCRIPTION
**Pull request by sonalibaskaran2499**
### Ticket
Link to Github Issue

### Problem description
Bringup BLOOM-176B model

### What's changed
Added loader for BLOOM-176B model.

### Logs
[bloom-176b.log](https://github.com/user-attachments/files/29293110/bloom-176b.log)

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: bloom/pytorch/loader.py [hunk 1]
```diff
@@ -132,9 +165,57 @@ def decode_output(self, outputs):
...
-        # Get logits for the last token in each batch
+            self.load_model()
         next_token_logits = outputs.logits[:, -1]
         next_tokens = next_token_logits.softmax(dim=-1).argmax(dim=-1)
         return [self.tokenizer.decode([token.item()]) for token in next_tokens]
+
+    def get_mesh_config(self, num_devices: int):
+        """Return mesh shape and axis names for tensor parallel."""
+        if num_devices == 32:  # Galaxy
+            mesh_shape = (8, 4)
```
Comment (kmabeeTT) [inline bloom/pytorch/loader.py]: Other recent galaxy models used mesh_shape = (4,8) for model axis 8 - just double checking if this is intentional here?

File: bloom/pytorch/loader.py [hunk 2]
```diff
@@ -91,32 +122,34 @@ def load_inputs(self, dtype_override=None, batch_size=1):
...
+        if self._variant == ModelVariant.BLOOM_176B:
+            inputs = self.tokenizer(
+                "This is a sample text to test the BLOOM model.",
+                return_tensors="pt",
+                padding=True,
+                truncation=True,
+                max_length=256,
+            )
+        else:
+            self.test_input = "This is a sample text from "
```
Comment (kmabeeTT) [inline bloom/pytorch/loader.py]: Duplicate, already set in __init__()

File: bloom/pytorch/loader.py [hunk 3]
```diff
@@ -91,32 +122,34 @@ def load_inputs(self, dtype_override=None, batch_size=1):
...
-                dtype_override=dtype_override
-            )  # This will initialize the tokenizer
-
-        # Create batch of sample inputs
-        self.test_input = "This is a sample text from "
+            self.load_model(dtype_override=dtype_override)
+
+        if self._variant == ModelVariant.BLOOM_176B:
+            inputs = self.tokenizer(
+                "This is a sample text to test the BLOOM model.",
```
Comment (kmabeeTT) [inline bloom/pytorch/loader.py]: Just use self.test_input already set in __init__() ?

**Review by kmabeeTT — APPROVED** [review]
Few small nits

File: bloom/pytorch/loader.py (hunk 3 above)
Comment (sonalibaskaran2499) [reply bloom/pytorch/loader.py]: Addressed all the review comments.
